doc/nrf/config_and_build/configuring_app/cmake/index.rst

```diff
@@ -44,24 +44,301 @@ See :ref:`zephyr:zephyr-app-cmakelists` in the Zephyr documentation for more inf
     This tutorial however differs from Zephyr and |NCS| project configurations, so use it only as reference.
 
 .. _cmake_options:
+.. _building_overlay_files:
 
 Providing CMake options
 ***********************
 
 You can provide additional options for building your application to the CMake process, which can be useful, for example, to switch between different build scenarios.
 These options are specified when CMake is run, thus not during the actual build, but when configuring the build.
 
-For information about what variables can be set and how to add these variables in your project, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
+The |NCS| uses the same CMake build variables as Zephyr and they are compatible with both CMake and west.
+For the complete list of build variables in Zephyr and more information about them, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
+The following table lists the most common ones used in the |NCS|:
+
+.. list-table:: Build system variables in the |NCS|
+   :header-rows: 1
+
+   * - Variable
+     - Purpose
+     - CMake argument to use
+   * - Name of the Kconfig option
+     - Set the given Kconfig option to a specific value :ref:`for a single build <configuration_temporary_change_single_build>`.
+     - ``-D<name_of_Kconfig_option>=<value>``
+   * - :makevar:`EXTRA_CONF_FILE`
+     - Provide additional :ref:`Kconfig fragment files <configuration_permanent_change>`.
+     - ``-DEXTRA_CONF_FILE=<file_name>.conf``
+   * - :makevar:`EXTRA_DTC_OVERLAY_FILE`
+     - Provide additional, custom :ref:`devicetree overlay files <configuring_devicetree>`.
+     - ``-DEXTRA_DTC_OVERLAY_FILE=<file_name>.overlay``
+   * - :makevar:`SHIELD`
+     - Select one of the supported :ref:`shields <shield_names_nrf>` for building the firmware.
+     - ``-DSHIELD=<shield_build_target>``
+   * - :makevar:`CONF_FILE`
+     - Select one of the available :ref:`build types <modifying_build_types>`, if the application or sample supports any.
+     - ``-DCONF_FILE=prj_<build_type_name>.conf``
+   * - ``-S`` or :makevar:`SNIPPET`
```

Comment:
Strictly speaking, `-S` isn't a CMake option but an argument to west. Perhaps mention instead that when using `west`, it's more common to specify snippets with `-S` instead of using `-DSNIPPET`?